aws-cpp-sdk-transfer/source/transfer/TransferHandle.cpp

```diff
@@ -238,17 +239,16 @@ namespace Aws
                         CleanupDownloadStream();
                     }
 
-                    std::unique_lock<std::mutex> semaphoreLock(m_statusLock);
                     m_waitUntilFinishedSignal.notify_all();
                 }
             }
         }
 
         void TransferHandle::WaitUntilFinished() const
         {
+            std::unique_lock<std::mutex> semaphoreLock(m_statusLock);
             if (!IsFinishedStatus(static_cast<TransferStatus>(m_status.load())) || HasPendingParts())
             {
-                std::unique_lock<std::mutex> semaphoreLock(m_statusLock);
                 m_waitUntilFinishedSignal.wait(semaphoreLock, [this]()
                     { return IsFinishedStatus(static_cast<TransferStatus>(m_status.load())) && !HasPendingParts(); });
                 semaphoreLock.unlock();
```

Comment:
Is it necessary to unlock explicitly? Shouldn't the system unlock it when lock get destructed?